module/mempool/stdmap/incorporated_result_seals.go

```diff
@@ -9,18 +11,124 @@ import (
 // of the consensus nodes, used to store seals that need to be added to blocks.
 type IncorporatedResultSeals struct {
 	*Backend
+	// index the seals by the height of the executed block
+	byHeight     map[uint64][]flow.Identifier
+	lowestHeight uint64
+}
+
+func indexByHeight(seal *flow.IncorporatedResultSeal) uint64 {
+	return seal.Header.Height
 }
 
 // NewIncorporatedResults creates a mempool for the incorporated result seals
 func NewIncorporatedResultSeals(opts ...OptionFunc) *IncorporatedResultSeals {
-	return &IncorporatedResultSeals{
-		Backend: NewBackend(opts...),
+	r := &IncorporatedResultSeals{
+		Backend:  NewBackend(opts...),
+		byHeight: make(map[uint64][]flow.Identifier),
+	}
+
+	// assuming all the entities are for unsealed blocks, then we will remove a seal
+	// with the highest height.
+	r.eject = func(entities map[flow.Identifier]flow.Entity) (flow.Identifier, flow.Entity) {
```

Comment:
If `opts` already contains an ejection logic, we are just overriding it here (without any error). I personally find that this is an unsafe API design. ATM, we are only supporting two `OptionFunc`:
* ejection 
* capacity limitation 

As we have hard-wired the ejection logic in this implementation, I think we can just use the capacity direclty as constructor argument: 
```golang
func NewIncorporatedResultSeals(limit uint) *IncorporatedResultSeals
```